packages/react/src/app-layout/AppLayoutHeader.tsx

```diff
@@ -32,6 +34,10 @@ export type AppLayoutHeaderProps = {
 	/** The logo to display. */
 	logo: JSX.Element;
 	palette?: ResponsiveProp<BoxPalette>;
+	/** The href to link to, for example "/". */
+	secondHref?: string;
+	/** The second logo to display for co-branding. */
+	secondLogo?: JSX.Element;
```

Comment:
Interesting. These header components are the only place we use `JSX.Element`. Suggest changing to `ReactElement` https://www.totaltypescript.com/jsx-element-vs-react-reactnode

Reply:
If we change the `logo` it'll be a breaking change. If we change this one then they're inconsistent. I feel we're stuck with `JSX.Element` here.

Reply:
They're functionally the same though, so it shouldn't matter?

Reply:
Ah yeah, they are functionally the same and so are interchangeable. I never realised. Why would you use one over the other, though?

Reply:
Reading that article, it feels lije `JSX.Element` was part of their early attempt at making TS compatible with JSX, and as things have progressed, they've been a bit less figure out "JSX and TS" and more "React and TS" is actually very compatible now, so let's lean more into using React in the types

Reply:
- [ ] Update to `ReactElement`